// @react-spectrum/layout/dist/Grid.main.js
var $ATZAs$reactspectrumutils = require("@react-spectrum/utils");
var $ATZAs$reactariautils = require("@react-aria/utils");
var $ATZAs$react = require("react");


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$export(module.exports, "Grid", () => $65aafe8662d00781$export$ef2184bd89960b14);
$parcel$export(module.exports, "repeat", () => $65aafe8662d00781$export$76d90c956114f2c2);
$parcel$export(module.exports, "minmax", () => $65aafe8662d00781$export$9c1b655deaca4988);
$parcel$export(module.exports, "fitContent", () => $65aafe8662d00781$export$2f0b47b0911ce698);
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 


const $65aafe8662d00781$var$gridStyleProps = {
    ...(0, $ATZAs$reactspectrumutils.baseStyleProps),
    autoFlow: [
        'gridAutoFlow',
        (0, $ATZAs$reactspectrumutils.passthroughStyle)
    ],
    autoColumns: [
        'gridAutoColumns',
        $65aafe8662d00781$var$gridDimensionValue
    ],
    autoRows: [
        'gridAutoRows',
        $65aafe8662d00781$var$gridDimensionValue
    ],
    areas: [
        'gridTemplateAreas',
        $65aafe8662d00781$var$gridTemplateAreasValue
    ],
    columns: [
        'gridTemplateColumns',
        $65aafe8662d00781$var$gridTemplateValue
    ],
    rows: [
        'gridTemplateRows',
        $65aafe8662d00781$var$gridTemplateValue
    ],
    gap: [
        'gap',
        (0, $ATZAs$reactspectrumutils.dimensionValue)
    ],
    rowGap: [
        'rowGap',
        (0, $ATZAs$reactspectrumutils.dimensionValue)
    ],
    columnGap: [
        'columnGap',
        (0, $ATZAs$reactspectrumutils.dimensionValue)
    ],
    justifyItems: [
        'justifyItems',
        (0, $ATZAs$reactspectrumutils.passthroughStyle)
    ],
    justifyContent: [
        'justifyContent',
        (0, $ATZAs$reactspectrumutils.passthroughStyle)
    ],
    alignItems: [
        'alignItems',
        (0, $ATZAs$reactspectrumutils.passthroughStyle)
    ],
    alignContent: [
        'alignContent',
        (0, $ATZAs$reactspectrumutils.passthroughStyle)
    ]
};
const $65aafe8662d00781$export$ef2184bd89960b14 = /*#__PURE__*/ (0, $ATZAs$react.forwardRef)(function Grid(props, ref) {
    let { children: children, ...otherProps } = props;
    let { styleProps: styleProps } = (0, $ATZAs$reactspectrumutils.useStyleProps)(otherProps, $65aafe8662d00781$var$gridStyleProps);
    if (styleProps.style) styleProps.style.display = 'grid'; // inline-grid?
    let domRef = (0, $ATZAs$reactspectrumutils.useDOMRef)(ref);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($ATZAs$react))).createElement("div", {
        ...(0, $ATZAs$reactariautils.filterDOMProps)(otherProps),
        ...styleProps,
        ref: domRef
    }, children);
});
function $65aafe8662d00781$export$76d90c956114f2c2(count, repeat) {
    return `repeat(${count}, ${$65aafe8662d00781$var$gridTemplateValue(repeat)})`;
}
function $65aafe8662d00781$export$9c1b655deaca4988(min, max) {
    return `minmax(${$65aafe8662d00781$var$gridDimensionValue(min)}, ${$65aafe8662d00781$var$gridDimensionValue(max)})`;
}
function $65aafe8662d00781$export$2f0b47b0911ce698(dimension) {
    return `fit-content(${$65aafe8662d00781$var$gridDimensionValue(dimension)})`;
}
function $65aafe8662d00781$var$gridTemplateAreasValue(value) {
    return value.map((v)=>`"${v}"`).join('\n');
}
function $65aafe8662d00781$var$gridDimensionValue(value) {
    if (/^max-content|min-content|minmax|auto|fit-content|repeat|subgrid/.test(value)) return value;
    return (0, $ATZAs$reactspectrumutils.dimensionValue)(value);
}
function $65aafe8662d00781$var$gridTemplateValue(value) {
    if (Array.isArray(value)) return value.map($65aafe8662d00781$var$gridDimensionValue).join(' ');
    return $65aafe8662d00781$var$gridDimensionValue(value);
}
